includes/html/api_functions.inc.php

```diff
@@ -1132,6 +1132,28 @@ function get_port_stack(Illuminate\Http\Request $request)
     });
 }
 
+function update_device_port_notes(Illuminate\Http\Request $request): \Illuminate\Http\JsonResponse
+{
+    $portid = $request->route('portid');
+
+    $hostname = $request->route('hostname');
+    // use hostname as device_id if it's all digits
+    $device_id = ctype_digit($hostname) ? $hostname : getidbyname($hostname);
+
+    $data = json_decode($request->getContent(), true);
+    $field = 'notes';
+    $content = $data[$field];
+    if (empty($data)) {
+        return api_error(400, 'Port field to patch has not been supplied.');
+    }
+
+    if (set_dev_attrib($device_id, 'port_id_notes:' . $portid, $content)) {
```

Comment:
does this work? never used this function

Reply:
Hey there @Jellyfrog, it works from my testing. I reused it from the functionality to modify the port notes in the UI here: https://github.com/librenms/librenms/blob/7c157c1577c7d7dceb01269114f100d08609b911/includes/html/forms/update-port-notes.inc.php#L21